docs/source/scope_tutorial.ipynb

```diff
@@ -0,0 +1,215 @@
+    "To remove the unused name, we implement ``RemoveUnusedImportTransformer`` by subclassing :class:`~libcst.CSTTransformer`. We overwrite ``leave_Import`` and ``leave_ImportFrom`` to modify the import statements.\n",
+    "When we find the import node in lookup table, we iterate through all ``names`` and keep used names in ``names_to_keep``.\n",
+    "If ``names_to_keep`` is empty, all names are unused and we remove the entire import node.\n",
+    "Otherwise, we update the import node and just removing partial names."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "class RemoveUnusedImportTransformer(cst.CSTTransformer):\n",
+    "    def __init__(\n",
+    "        self, unused_imports: Dict[Union[cst.Import, cst.ImportFrom], Set[str]]\n",
+    "    ) -> None:\n",
+    "        self.unused_imports = unused_imports\n",
+    "\n",
+    "    def leave_import_alike(\n",
+    "        self,\n",
+    "        original_node: Union[cst.Import, cst.ImportFrom],\n",
+    "        updated_node: Union[cst.Import, cst.ImportFrom],\n",
+    "    ) -> Union[cst.Import, cst.ImportFrom, cst.RemovalSentinel]:\n",
+    "        if original_node not in self.unused_imports:\n",
+    "            return updated_node\n",
+    "        names_to_keep = []\n",
+    "        for name in updated_node.names:\n",
+    "            asname = name.asname\n",
+    "            if asname is not None:\n",
+    "                name_value = asname.name.value\n",
+    "            else:\n",
+    "                name_value = name.name.value\n",
+    "            if name_value not in self.unused_imports[original_node]:\n",
+    "                names_to_keep.append(name.with_changes(comma=cst.MaybeSentinel.DEFAULT))\n",
+    "        if len(names_to_keep) == 0:\n",
+    "            return cst.RemoveFromParent()\n",
+    "        else:\n",
+    "            return updated_node.with_changes(names=names_to_keep)\n",
+    "\n",
+    "    def leave_Import(\n",
+    "        self, original_node: cst.Import, updated_node: cst.Import\n",
+    "    ) -> cst.Import:\n",
+    "        return self.leave_import_alike(original_node, updated_node)\n",
+    "\n",
+    "    def leave_ImportFrom(\n",
+    "        self, original_node: cst.ImportFrom, updated_node: cst.ImportFrom\n",
+    "    ) -> cst.ImportFrom:\n",
+    "        return self.leave_import_alike(original_node, updated_node)\n"
+   ]
+  },
+  {
+   "cell_type": "raw",
+   "metadata": {
+    "raw_mimetype": "text/restructuredtext"
+   },
+   "source": [
+    "After the transform, we use ``.code`` to generate fixed code and all unused names are fixed as expected!"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "fixed_module = wrapper.module.visit(RemoveUnusedImportTransformer(unused_imports))\n",
+    "print(fixed_module.code)"
```

Comment:
You could probably use difflib here to make the transformation more obvious, similar to what you did in the first tutorial: https://libcst.readthedocs.io/en/latest/tutorial.html#Generate-Source-Code